goal_src/jak1/pc/features/speedruns.gc

```diff
@@ -1,41 +1,254 @@
 ;;-*-Lisp-*-
 (in-package goal)
 
-(defun speedrun-mode-update ()
-  "A per frame update for speedrunning related stuff"
-  (when (-> *pc-settings* speedrunner-mode?)
-    ;; Update auto-splitter
-    (update-autosplit-info-jak1)
-    ;; Draw info to the screen
-    (speedrun-draw-settings))
-  (none))
+(define *speedrun-info* (new 'static 'speedrun-info-jak1 :should-display? #t :needs-post-blackout-setup? #f))
 
-(defun speedrun-start-run ()
-  ;; randomize game id so the autosplitter knows to restart
+(defun is-speedrun-category? ((category symbol))
+  (and (-> *pc-settings* speedrunner-mode?)
+        (name= category (enum->string speedrun-category (-> *speedrun-info* category)))
+        )
+  )
+
+(defun start-speedrun ((category speedrun-category));; randomize game id so the autosplitter knows to restart
   (update-autosplit-jak1-new-game)
-  ;; spawn at the warp gate checkpoint
-  (initialize! *game-info* 'game (the-as game-save #f) "game-start")
-  ;; disable hints (this seems to be overriden by your slot 1 save though)
+  ;; disable hints
   (set! (-> *setting-control* default play-hints) #f)
```

Comment:
Moving this up before the `initialize!` call seems to properly disable play-hints, regardless of previous save's play-hint setting.

Reply:
Yeah I did this in my PR as well, from my testing it works no matter what

Reply:
If you reboot in debug mode and start a speedrun from the new hidden menu, play-hints was left on (without this reordering)